internal/rpc/client/token.go

```diff
@@ -0,0 +1,23 @@
+package client
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/twitchtv/twirp"
+
+	"github.com/aquasecurity/trivy/pkg/log"
+)
+
+func WithToken(ctx context.Context, token string) context.Context {
+	// Prepare custom header
+	header := make(http.Header)
+	header.Set("Trivy-Token", token)
+
+	// Attach the headers to a context
+	ctx, err := twirp.WithHTTPRequestHeaders(ctx, header)
```

Comment:
If `twirp.WithHTTPRequestHeaders` returns an error, `ctx` will be nil, which could potentially blow up the callers of this method that rely on the returned `ctx` value.

Reply:
One example of how this will blow up is this: https://github.com/twitchtv/twirp/blob/master/context.go#L93-L96 which is currently used to test the returned headers. Maybe we can return a context with cancelation instead of nil? Ideally this should what the library should've done IMO.

Reply:
WithHTTPRequestHeaders returns an error only when a reserved header is used. In this case, this function doesn't return an error. But as you said, it's not fine to return `nil`.
https://github.com/twitchtv/twirp/blob/master/context.go#L69

What does it mean that "we can return a context with cancelation"? Do you mean `context.WithCancel(ctx)`?

I'm thinking of the following way.

```
ctxWithToken, err := twirp.WithHTTPRequestHeaders(ctx, header)
if err != nil {
	log.Logger.Warnf("twirp error setting headers: %s", err)
	return ctx // this is not nil
}
return ctxWithToken
```

Reply:
Yes that's what I mean: `context.WithCancel(ctx)`. Then the downstream receivers of this context can cancel their operations if they choose to. You can read more on this here https://www.sohamkamani.com/blog/golang/2018-06-17-golang-using-context-cancellation/ 

What you have suggested is OK too, it just doesn't utilize the canceling functionality downstream.

Reply:
I added a commit to handle this: https://github.com/aquasecurity/trivy/pull/302/commits/d63f8b2dd7e12e21795e0471fc1a00dd95ec540d